src/DssSpell.t.base.sol

```diff
@@ -1820,52 +1808,175 @@ contract DssSpellTestBase is Config, DssTest {
+                diffCount += (cacheAfter.count - cacheBefore.count);
+            }
+
+            ////////////////////////////////////////
+
+            bytes32[] memory diffKeys = new bytes32[](diffCount);
+            uint256 j = 0;
 
-    // Ensure version is updated if chainlog changes
-    function _testChainlogVersionBump() internal {
-        uint256                   _count = chainLog.count();
-        string    memory        _version = chainLog.version();
-        address[] memory _chainlog_addrs = new address[](_count);
+            for (uint256 i = 0; i < maxIters; i++) {
+                if (cacheAfter.keys[i] != cacheBefore.keys[i]) {
+                    // Mark keys whose order has changed
+                    diffKeys[j++] = cacheAfter.keys[i];
+                } else if (cacheAfter.values[i] != cacheBefore.values[i]) {
+                    // Mark changed values
+                    diffKeys[j++] = cacheAfter.keys[i];
+                }
+            }
+
+            // Mark new keys
+            if (cacheAfter.count > cacheBefore.count) {
+                for (uint256 i = cacheBefore.count; i < cacheAfter.count; i++) {
+                    diffKeys[j++] = cacheAfter.keys[i];
+                }
+            }
 
-        for(uint256 i = 0; i < _count; i++) {
-            (, address _val) = chainLog.get(i);
-            _chainlog_addrs[i] = _val;
+            for (uint256 i = 0; i < diffKeys.length; i++) {
+                _checkAuth(diffKeys[i]);
+            }
         }
+    }
 
+    // Validate addresses in test harness match chainlog
+    function _testChainlogValues() internal {
         _vote(address(spell));
         _scheduleWaitAndCast(address(spell));
         assertTrue(spell.done());
 
-        if (keccak256(abi.encodePacked(_version)) == keccak256(abi.encodePacked(chainLog.version()))) {
-            // Fail if the version is not updated and the chainlog count has changed
-            if (_count != chainLog.count()) {
-                emit log_named_string("Error", _concat("TestError/chainlog-version-not-updated-count-change-", _version));
-                fail();
-                return;
-            }
-            // Fail if the chainlog is the same size but local keys don't match the chainlog.
-            for(uint256 i = 0; i < _count; i++) {
-                (, address _val) = chainLog.get(i);
-                if (_chainlog_addrs[i] != _val) {
-                    emit log_named_string("Error", _concat("TestError/chainlog-version-not-updated-address-change-", _version));
-                    fail();
-                    return;
-                }
-            }
+        bytes32[] memory keys = chainLog.list();
+        for (uint256 i = 0; i < keys.length; i++) {
+            assertEq(
+                chainLog.getAddress(keys[i]),
+                addr.addr(keys[i]),
+                _concat("TestError/chainlog-vs-harness-key-mismatch: ", keys[i])
+            );
```

Comment:
considering the PR objective was initially to reduce testing times, this check could be done inside `_testChainlogIntegrity`, when creating the `cacheAfter` object, optimizing 1/3 of the calls, since it'll have to query all chainlog values only twice (before spell and after spell)